### ISSUE TITLE
Null ref in CommitSolutionUpdate

### ISSUE THREAD

**Issue by pranavkm** [dotnet/roslyn#56617]
VS version: Dev 17 Preview 4

```
StreamJsonRpc.RemoteInvocationException: Unexpected null - line 225
at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__139`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Threading.Tasks.ValueTask`1.get_Result()
at Microsoft.CodeAnalysis.Remote.BrokeredServiceConnection`1.<TryInvokeAsync>d__12`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
RPC server exception:
System.InvalidOperationException: Unexpected null - line 225
at Roslyn.Utilities.Contract.Fail(String message, Int32 lineNumber)
at Roslyn.Utilities.Contract.ThrowIfNull[T](T value, Int32 lineNumber)
at Microsoft.CodeAnalysis.EditAndContinue.EditAndContinueWorkspaceService.CommitSolutionUpdate(DebuggingSessionId sessionId, ImmutableArray`1& documentsToReanalyze)
at Microsoft.CodeAnalysis.EditAndContinue.RemoteEditAndContinueService.<>c__DisplayClass12_0.<CommitSolutionUpdateAsync>b__0(CancellationToken cancellationToken)
at Microsoft.CodeAnalysis.Remote.BrokeredServiceBase.<RunServiceImplAsync>d__12`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

It happened as part of launching the project before any hot reload

**Comment by pranavkm** [comment]
FYI @tmat

**Comment by tmat** [comment]
@pranavkm Have a repro or memory dump?

**Comment by pranavkm** [comment]
We were trying to use hot reload on https://github.com/OrchardCMS/OrchardCore/blob/dotnet6/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj with 17.0 p4. @sebastienros in case you're able to still reproduce this and are able to provide @tmat a dump.

**Comment by isadorasophia** [comment]
I think I might see what's the race condition on the debugger side. It sounds like we are commiting changes without ever really starting the hot reload session. We should first await for any pending start notification events to the language service before proceeding with any apply changes.

**Comment by tmat** [comment]
I believe we have addressed these race conditions in 17.0 and 17.1.
